server/repository/src/db_diesel/assets/asset_log_row.rs

```diff
@@ -1,36 +1,97 @@
 use super::asset_log_row::asset_log::dsl::*;
 
+use crate::EqualFilter;
 use crate::RepositoryError;
 use crate::StorageConnection;
 
 use chrono::NaiveDateTime;
 use diesel::prelude::*;
 
+use diesel_derive_enum::DbEnum;
+use serde::{Deserialize, Serialize};
+
 table! {
     asset_log (id) {
         id -> Text,
         asset_id -> Text,
         user_id -> Text,
-        status -> Nullable<Text>,
+        status -> Nullable<crate::db_diesel::assets::asset_log_row::StatusMapping>,
         comment -> Nullable<Text>,
         #[sql_name = "type"] type_ -> Nullable<Text>,
-        reason -> Nullable<Text>,
+        reason -> Nullable<crate::db_diesel::assets::asset_log_row::ReasonMapping>,
         log_datetime -> Timestamp,
     }
 }
 
+#[derive(DbEnum, Debug, Clone, PartialEq, Eq, Serialize, Deserialize)]
+#[serde(rename_all = "SCREAMING_SNAKE_CASE")]
+#[DbValueStyle = "SCREAMING_SNAKE_CASE"]
+
+pub enum Reason {
+    AwaitingInstallation,
+    Stored,
+    OffsiteForRepairs,
+    AwaitingDecomissioning,
+    NeedsServicing,
+    MultipleTemperatureBreaches,
```

Comment:
These look very cold chain specific are they not customer defined/configured ?

Reply:
I agree it feels a bit arbitrary and customer specific. I wonder if it would be better to leave it as a configurable enum and remove the matching-reason-to-status validation. 

@richardmoizeau do you have any thoughts on if the reasons and status's in https://docs.google.com/spreadsheets/d/1l1x38KT5cdt4Zd5piMWp3FuBuiSUv4faMHP8utDJDqU/edit#gid=2113323501 would be configurable?

Reply:
Discussed with @mark-prins and @jmbrunskill - we will add as is with option to make configurable as a future issue if required